### PR TITLE
V5 client version

### PR DESCRIPTION
**Pull request by romanyx**
Client prepared for v5 release. https://github.com/messagebird/go-rest-api/issues/34

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -63,118 +60,26 @@ type Client struct {
...
-	// We're dealing with an API error here. try to decode it, but don't do
-	// anything with the error. This is because not all values of `v` have
-	// `Error` properties and decoding could fail.
-	json.Unmarshal(responseBody, &v)
-
-	// Anything else than a 200/201/500 should be a JSON error.
-	return ErrResponse
+// NewV2 creates a new MessageBird client object.
+// Deprecated: use New factory instead.
+func NewV2(accessKey string) *Client {
```
Comment (epels) [inline client.go]: Please remove `NewV2` - no need for deprecating it as we're releasing this with a major version bump

**Comment by epels** [comment]
Also: could you please make sure all `Errors` fields are removed from the struct definitions? Thanks in advance!

**Comment by romanyx** [comment]
Hi! I did the changes what you have been requested, looks like the request is ready for merge.
Totally agree with you that the error should be as more concise as it can. I kept error text capitalized, as same as it's in `ErrUnexpectedResponse` although this violates the [errors string convention](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).
Should I squash commits into the single one?

**Comment by epels** [comment]
Hi @romanyx, thanks a lot for making the changes so quickly! It looks like the [`VoiceMessage` struct](https://github.com/romanyx/go-rest-api/blob/refactor/client_v2/voice_message.go#L23) still has the `[]Error` field - after that is removed, we're indeed good to merge this!

**Comment by romanyx** [comment]
Oh, search somehow didn't find this sneaky one.

**Comment by romanyx** [comment]
Forgot to change the test, should work now.

**Comment by epels** [comment]
@romanyx Great - thanks a bunch for your much appreciated contribution!

We're planning on releasing the new version next week. I have one final request: could you please change the target branch to [`v5_staging`](https://github.com/messagebird/go-rest-api/tree/v5_staging)? We're making a few more changes, and it's easiest for me to have them all on a single branch before releasing. Thanks in advance!

**Pull request merged**